pkg/webhook/admission/pod/storage_initializer_injector.go

```diff
@@ -330,12 +331,26 @@ func (mi *StorageInitializerInjector) InjectStorageInitializer(pod *v1.Pod) erro
 		}
 	}
 
-	// Allow to override the uid for the case where ISTIO CNI with DNS proxy is enabled
-	// See for more: https://istio.io/latest/docs/setup/additional-setup/cni/#compatibility-with-application-init-containers.
-	if value, ok := pod.GetAnnotations()[constants.IstioSidecarUIDAnnotationKey]; ok {
-		if uid, err := strconv.ParseInt(value, 10, 64); err == nil {
-			initContainer.SecurityContext.RunAsUser = ptr.Int64(uid)
-		}
+	/*
+		OpenShift uses istio-cni which causes an issue with init-containers when calling external services
+		like S3 or similar. Setting the `uid` for the `storage-initializer` to the same `uid` as the
+		`uid` of the `istio-proxy` resolves the issue. In OpenShift the `istio-proxy` always gets assigned
+		the first `uid` from the namespaces `uid` range + 1 (The range is defined in an annotation on the namespace).
+	*/
+	if initContainer.SecurityContext == nil {
+		initContainer.SecurityContext = &v1.SecurityContext{}
+	}
+	uidStr := targetNs.Annotations[OpenShiftUidRangeAnnotationKey]
+	if uidStr == "" {
+		return fmt.Errorf("could not find annotation %s on namespace: %s", OpenShiftUidRangeAnnotationKey, targetNs.Name)
```

Comment:
Who is it adding this annotation? Istio? Can you add this context to the error message?

Reply:
That is an annotation OpenShift adds to every namespace. But yes, I can add it to the message.